lib/SPIRV/SPIRVToOCL.cpp

```diff
@@ -515,6 +519,32 @@ void SPIRVToOCLBase::visitCallSPIRVImageMediaBlockBuiltin(CallInst *CI, Op OC) {
       &Attrs);
 }
 
+void SPIRVToOCLBase::visitCallGenericCastToPtrExplicitBuiltIn(CallInst *CI,
+                                                              Op OC) {
+  AttributeList Attrs = CI->getCalledFunction()->getAttributes();
+  mutateCallInstOCL(
+      M, CI,
+      [=](CallInst *Call, std::vector<Value *> &Args) {
+        auto AddrSpace = static_cast<SPIRAddressSpace>(
+            CI->getType()->getPointerAddressSpace());
+        // The instruction has two arguments, whereas ocl built-in has only one
+        // argument.
+        Args.pop_back();
+        switch (AddrSpace) {
+        case SPIRAS_Global:
+          return std::string(kOCLBuiltinName::ToGlobal);
+        case SPIRAS_Local:
+          return std::string(kOCLBuiltinName::ToLocal);
+        case SPIRAS_Private:
+          return std::string(kOCLBuiltinName::ToPrivate);
+        default:
+          llvm_unreachable("Invalid address space");
+          return std::string();
```

Comment:
To fulfill my curiosity: compiler emits a warning if we don't add a return here?

Reply:
Yes.